android/FacebookPlugin.java

```diff
@@ -444,6 +450,90 @@ public void logout(String json, Integer requestId) {
     sendResponse(res, null, requestId);
   }
 
+  public void logEvent (String s_json) {
+    logger.log("{facebook} logEvent");
+
+    JSONObject json        = null;
+    String     eventName   = null;
+    Double     valueToSum  = null;
+    JSONObject _parameters = null;
+    Bundle     parameters  = null;
+
+    // Parse JSON;
+    try {
+      json        = new JSONObject(s_json);
+      eventName   = json.optString("eventName");
+      valueToSum  = json.optDouble("valueToSum", 0);
+      _parameters = json.optJSONObject("parameters");
+    } catch (JSONException e) {
+      onJSONException(e);
+      log("logEvent failed to parse JSON blob");
+      return;
+    }
+
+    // Convert JSONObject _parameters into a Bundle
+    if (_parameters.length() > 0) {
+      try {
+        parameters = BundleJSONConverter.convertToBundle(_parameters);
+      } catch (JSONException e) {
+        log("logEvent - error converting JSONObject to Bundle");
+      }
+    }
+
+    // Log the event
+    if(valueToSum != null && valueToSum != 0){
+      if(parameters != null){
+        aeLogger.logEvent(eventName, valueToSum, parameters);
+      } else {
+        aeLogger.logEvent(eventName, valueToSum);
+      }
+    } else {
+      if(parameters != null){
+        aeLogger.logEvent(eventName, parameters);
+      } else {
+        aeLogger.logEvent(eventName);
+      }
+    }
+  }
+
+  public void logPurchase (String s_json) {
+    logger.log("{facebook} logPurchase");
+
+    JSONObject json           = null;
+    Double     purchaseAmount = null;
+    String     currency       = null;
+    JSONObject _parameters    = null;
+    Bundle     parameters     = null;
+
+    // Parse JSON;
+    try {
+      json           = new JSONObject(s_json);
+      purchaseAmount = json.optDouble("purchaseAmount");
+      currency       = json.optString("currency");
+      _parameters    = json.optJSONObject("parameters");
+    } catch (JSONException e) {
+      onJSONException(e);
+      log("logEvent failed to parse JSON blob");
+      return;
+    }
+
+    // Convert JSONObject _parameters into a Bundle
+    if (_parameters.length() > 0) {
+      try {
+        parameters = BundleJSONConverter.convertToBundle(_parameters);
+      } catch (JSONException e) {
+        log("logPurchase - error converting JSONObject to Bundle");
+      }
+    }
+
+    // Log the purchase
+    if(parameters != null){
+      aeLogger.logPurchase(BigDecimal.valueOf(purchaseAmount), Currency.getInstance(currency), parameters);
+    } else {
+      aeLogger.logPurchase(BigDecimal.valueOf(purchaseAmount), Currency.getInstance(currency));
+    }
```

Comment:
You can just pass parameters in whether or not it's null which means this check it not needed at all on android. [link](https://github.com/facebook/facebook-android-sdk/blob/master/facebook/src/com/facebook/appevents/AppEventsLogger.java#L544-L546).